mirascope/tools/search/duckduckgo.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from typing import ClassVar, TypeVar
+
+from duckduckgo_search import DDGS
+from pydantic import Field
+
+from ..base import _ToolConfig
+from .base import SearchToolBase
+
+
+class DuckDuckGoSearchConfig(_ToolConfig):
+    """Configuration for DuckDuckGo search"""
+
+    max_results_per_query: int = Field(
+        default=2, description="Maximum number of results per query"
+    )
+
+    @classmethod
+    def from_env(cls) -> _ToolConfig:
+        return cls()
+
+
+_ToolSchemaT = TypeVar("_ToolSchemaT")
+
+
+class DuckDuckGoSearch(SearchToolBase[DuckDuckGoSearchConfig, _ToolSchemaT]):
+    """Tool for performing web searches using DuckDuckGo.
+    Takes search queries and returns relevant URLs from search results.
+    """
+
+    __config__ = DuckDuckGoSearchConfig()
+    __prompt_usage_description__: ClassVar[str] = """
+    Use this tool to search the web and get relevant URLs.
+
+    Enter search queries to get URLs from search results.
+    """
+
+    queries: list[str] = Field(..., description="List of search queries")
+
+    def call(self) -> str:
+        try:
+            urls = []
+            for query in self.queries:
+                results = DDGS(proxies=None).text(
```

Comment:
realizing here that we somehow need to support both sync and async, particularly for tools with I/O bound operations.

Generally with `BaseTool` the user would just define `call` as async and call it a day, so for pre-made tools it seems like the best option would be to implement an async version of the tools (e.g. `AsyncDuckDuckGoSearch`)

WDYT?

Reply:
Makes sense.I will add it soon.